### ISSUE TITLE
"OK" button changes behavior

### ISSUE THREAD

**Issue by horomancer** [sspanak/tt9#568]
I'm using a stock Cat S22 flip phone and Traditional T9 was a great asset to text without taking up half my screen with a virtual keyboard.
When first installed it's default behavior was to send messages when pressing the "OK" button, however this changed to making a new line in the text field, similar to pressing Shift+Enter on a keyboard. This changed happened once after installing Voice Access, as I was trying out different control methods. I uninstall and reinstalled TT9 and the problem went away. It came back again after I tried different launcher programs to customize my phone's looks. Unlike before, uninstalling and reinstalling  TT9 does not correct the issue. This change in behavior effected both the default Messages app and Google Voice texting. Neither of those apps have updated or had their settings altered.
I tried setting the "Set 'OK' to send..." options to ON in the TT9 settings, but that did not effect anything.

TT9 has been the best looking and most effect T9 keyboard app. I really would love for this issue to be figured out, since the 'new line' behavior makes a lot of friction to send text messages compared to the 'send'. I see in the documentation this behavior should be controlled by the messaging apps, but i think it's weird that those apps haven't changed and functioned in the preferred method, then changed when 3rd party apps that may interact with them got introduced.

let me know if there is anything i can do on my end to give you more info.

**Comment by sspanak** [comment]
The logic for sending messages (or performing the default app action for that matter) hasn't changed since the beginning of April. Some other change might have caused a side effect, but I highly doubt that, because it would have affected all phones, not just the Cat S22.

However, this:
>  It came back again after I tried different launcher programs to customize my phone's looks. Unlike before, uninstalling and reinstalling TT9 does not correct the issue.

... seems like a plausible reason. The launcher apps may take precedence over the keyboards and modify the key press information before passing it to another application. Please, disable all custom launchers and try again with the stock one.

> i think it's weird that those apps haven't changed and functioned in the preferred method, then changed when 3rd party apps that may interact with them got introduced.

Yes, it looks weird, I can imagine. It's because in Android there are three different versions of the "OK" key. And each app is designed for a different "OK". TT9 supports all the hardware alternatives and tries to guess and send the correct "OK" depending on the app, but a launcher in the middle may make things difficult. This is why I am asking you to go back to the stock launcher and try again.

**Comment by horomancer** [comment]
I've uninstalled all launcher apps and TT9, then reinstalled TT9. Preferred functionality was restored. I then installed only the custom launcher I wanted to use and nothing broke. I have not tried reinstalling Voice Access and so far the function of the "OK" key has not changed and things have been stable.
Something i did not try, but will if things change again, is to switch keyboards from TT9 to Kika 12-Key. If Voice Access was somehow changing the behavior of the apps, perhaps there is a setting being effected in those apps by Voice Access. If so I would be able to see the function of Kika change, and if it doesn't change see if it is reverting the app settings and possibly fix TT9's functions.
